src/main/java/com/laytonsmith/core/events/EventUtils.java

```diff
@@ -181,12 +182,62 @@ public static void ManualTrigger(String eventName, CArray object, Target t, bool
 		}
 	}
 
+	private static Map<Event, Map<String, Prefilter<BindableEvent>>> prefilterData = new HashMap<>();
+
+	/**
+	 * This returns whether or not the given BoundEvent's prefilters matches the event. This method will never
+	 * throw a {@link PrefilterNonMatchException}
+	 * @param b
+	 * @param e
+	 * @param driver
+	 * @return
+	 */
+	public static boolean PrefilterMatches(BoundEvent b, BindableEvent e, Event driver) {
+		Map<String, Prefilter<BindableEvent>> prefilters;
+		// Keep a cache so we only build the prefilters object once.
+		if(!prefilterData.containsKey(driver)) {
+			PrefilterBuilder prefilterBuilder = driver.getPrefilters();
+			if(prefilterBuilder != null) {
+				prefilters = prefilterBuilder.build();
+			} else {
+				prefilters = null;
+			}
+			prefilterData.put(driver, prefilters);
+		} else {
+			prefilters = prefilterData.get(driver);
+		}
```

Comment:
Could each event's built prefilters be stored in AbstractEvent instead?

Reply:
Hmm, not without changing the Event interface again, or rather, changing all the subclasses. It's also ok for the cache to live outside the Event objects, so we don't violate the single responsibility principal too badly.

Reply:
What's wrong with changing the Event interface is if it's implemented in AbstractEvent?

Reply:
Alright, that cleaned things up a lot nicer actually, so good call.